sdk/data/azcosmos/cosmos_container.go

```diff
@@ -456,6 +457,66 @@ func (c *ContainerClient) NewQueryItemsPager(query string, partitionKey Partitio
 	})
 }
 
+// NewTransactionalBatch creates a batch of operations to be committed as a single unit.
+// See https://docs.microsoft.com/azure/cosmos-db/sql/transactional-batch
+func (c *ContainerClient) NewTransactionalBatch(partitionKey PartitionKey) TransactionalBatch {
+	return TransactionalBatch{partitionKey: partitionKey}
+}
+
+// ExecuteTransactionalBatch executes a transactional batch.
+// Once executed, verify the Success property of the response to determine if the batch was committed
+func (c *ContainerClient) ExecuteTransactionalBatch(ctx context.Context, b TransactionalBatch, o *TransactionalBatchOptions) (TransactionalBatchResponse, error) {
+	if len(b.operations) == 0 {
```

Comment:
The guidelines state that we should omit client-side validation of parameters and let the service reject the request.  But before we change it, I have some questions.

- Does the service return an intelligible error in this case?
- Would the caller be charged $$ for such a call?

Reply:
The service returns a BadRequest but not exactly the reason for the BadRequest, so I preferred to be more specific for this case.

Reply:
I see, we can leave it for now then.

Long-term, can this get fixed on the service side?  Then the error is uniform across all SDKs as well as customers that directly use the HTTP endpoints.

Reply:
That would be ideal, I agree. And as per previous PRs, I try to avoid any client side validation, but this case I feel customers would be confused.